kroxylicious-kms-providers/kroxylicious-kms-provider-azure-key-vault-kms/src/main/java/io/kroxylicious/kms/provider/azure/config/auth/ManagedIdentityConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.azure.config.auth;
+
+import java.net.URI;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+/**
+ * @param targetResource required App ID URI of the target resource
+ * @param identityServiceHost optional host address for Managed Identity Service (should only be configured for testing), defaults to "169.254.169.254" which is the Azure Instance Metadata Service (IMDS)
+ * @param identityServicePort optional port for Managed Identity Service (should only be configured for testing), defaults to null implying no port will be included in requests
+ */
+@JsonPropertyOrder({ "targetResource", "identityServiceHost", "identityServicePort" })
+public record ManagedIdentityConfig(@JsonProperty(required = true) String targetResource,
```

Comment:
non-blocking. I've realised the existing `EntraConfig` has a default value for it's equivalent `scope` property of `https://vault.azure.net/.default`. We should decide which way we prefer and do both the same. We could have a default of `https://vault.azure.net` here. Or make `scope` required on `EntraConfig`.

Maybe we can defer it to a subsequent PR with configuration changes as you were suggesting renaming `entraIdentity`.